x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/pki/PkiRealm.java

```diff
@@ -214,7 +214,8 @@ private void buildUser(X509AuthenticationToken token, String principal, ActionLi
                 "pki_delegated_by_user",
                 token.getDelegateeAuthentication().getEffectiveSubject().getUser().principal(),
                 "pki_delegated_by_realm",
-                token.getDelegateeAuthentication().getAuthenticatedBy().getName()
+                // TODO: this should be the realm of effective subject
+                token.getDelegateeAuthentication().getAuthenticatingSubject().getRealm().getName()
```

Comment:
This is a bug. I'll have a separate PR just for it so it is not blended in a pure refactoring.